core/src/main/java/com/yahoo/oak/EntryHashSet.java

```diff
@@ -185,7 +185,7 @@ private boolean isKeyAndEntryKeyEqual(KeyBuffer tempKeyBuff, K key, int idx, int
         }
 
         assert tempKeyBuff.isAssociated();
-        return (0 == comparator.compareKeyAndSerializedKey(key, tempKeyBuff));
+        return (0 == ((KeyMemoryManager ) keysMemoryManager).compareKeyAndSerializedKey(key, tempKeyBuff, comparator));
```

Comment:
I don't see why this is necessary? All MM that needs to lock the keys has the same implementation: lock / compare / unlock.
The only MM that is different is `SeqExpand`. And its lock/unlock methods do nothing anyway.

Reply:
I am not sure what is your question, 
are you asking why I am wrapping the compare with the memory manger lock/ unlock?
I am doing so, that when we have a concurrent delete with another access, we need to be notified about this.

Reply:
It is a statement, not a quesion.
I don't understand why each memory manager has to implement its own compare keys function.
(a) It is not modular. MM has nothing to do with key's comparison.
(b) It is not necessary. It can be implemented easily outside the MM.
Please modify your code such that these compare functions will be implemented once (not for each MM), and in another class (not inside MM).
I suggest maybe a new class for key management (like the one we have for values).